### PR TITLE
Rewards on utility process

### PR DESCRIPTION
**Pull request by yrliou**
Fix https://github.com/brave/brave-browser/issues/1708
Fix https://github.com/brave/brave-browser/issues/1709
## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Windows
  - [x] macOS
  - [x] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [x] Windows
  - [x] macOS
  - [x] Linux
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by yrliou** [comment]
I think this PR is ready for an initial review.
Rebase is needed again for this PR, I need to make those new getters added today async, also might have to revise some existing tests.

File: browser/importer/brave_profile_writer.cc [hunk 1]
```diff
@@ -300,9 +300,13 @@ void BraveProfileWriter::UpdateLedger(const BraveLedger& ledger) {
   }
 
   ledger_ = BraveLedger(ledger);
+  rewards_service_->IsWalletCreated(
```
Comment (bridiver) [inline browser/importer/brave_profile_writer.cc]: this just came up recently I think with @ryanml. We should be checking `rewards_service_->is_enabled()` - but this doesn't seem related to this PR?

Comment (bridiver) [reply browser/importer/brave_profile_writer.cc]: I see, it's because IsWalletCreated now has a callback. In either case we should always check to ensure that rewards service is enabled before calling any methods on it

**Comment by NejcZdovc** [comment]
@bridiver we should really make sure that this is not regressing in anyway as we are really close to stabilize master and would really like to keep it like this. Would love to get @LaurenWags's help on this PR as well

**Comment by yrliou** [comment]
Auto tests are now all passed on mac, also played for a bit on mac, try creating wallet, tipping, and visit sites & check contribution list, which so far seems fine to me.
I'm now building on all platforms to test on them too.

File: patches/chrome-browser-chrome_content_utility_manifest_overlay.json.patch [hunk 2]
```diff
@@ -1,12 +0,0 @@
-diff --git a/chrome/browser/chrome_content_utility_manifest_overlay.json b/chrome/browser/chrome_content_utility_manifest_overlay.json
-index 03d7cd6b486001765096df3bfe95624fd2700c42..a5dbd36e97f36417b861c41a2bfb201a8640aba2 100644
---- a/chrome/browser/chrome_content_utility_manifest_overlay.json
-+++ b/chrome/browser/chrome_content_utility_manifest_overlay.json
-@@ -10,6 +10,7 @@
-           "payments.mojom.PaymentManifestParser",
-           "heap_profiling.mojom.ProfilingClient",
-           "proxy_resolver.mojom.ProxyResolverFactory",
-+          "tor.mojom.TorLauncher",
```
Comment (NejcZdovc) [inline patches/chrome-browser-chrome_content_utility_manifest_overlay.json.patch]: should this be here?

Comment (yrliou) [reply patches/chrome-browser-chrome_content_utility_manifest_overlay.json.patch]: Nope, it's not a typo, we found this patch seems to be unnecessary since ads and rewards work fine with out patching here.

Comment (NejcZdovc) [reply patches/chrome-browser-chrome_content_utility_manifest_overlay.json.patch]: ok sounds good

**Comment by NejcZdovc** [comment]
master (0.61) 3a4c91a
0.60 46d16689
